server/src/main/java/org/elasticsearch/search/rank/rerank/ActionRequestRankFeaturePhaseRankCoordinatorContext.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.rank.rerank;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.search.rank.context.RankFeaturePhaseRankCoordinatorContext;
+import org.elasticsearch.search.rank.feature.RankFeatureDoc;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A global reranker that computes the updated scores for the top-K results from all shards, through an inference service.
+ * This coordinator context, that's executed at the end of the {@link org.elasticsearch.action.search.RankFeaturePhase},
+ * needs an {@code inferenceId} and {@code inferenceText} to generate the request to the inference service
+ * and then iterates over the results to update the scores in-place.
+ * The {@code Client} should ensure that this is done in an efficient non-blocking manner,
+ * and then pass the results back through the provided {@code ActionListener}.
+ * Each implementation of this class needs to define the request generation logic, the action type, and how to actually read the scores
+ * from the service's response.
+ */
+public abstract class ActionRequestRankFeaturePhaseRankCoordinatorContext<Request extends ActionRequest, Response extends ActionResponse>
```

Comment:
I don't understand the need for a new abstract class. To my knowledge, we will only have  1 maybe 2 async rankers. 

I think all of this should be encapsulated in the concrete classes. Adding this new abstraction doesn't make sense to me.

Reply:
> If you prefer to get rid of this addition, I can remove it from the PR and the concrete implementations can define it as needed.

It seems to me that this is providing a way for multiple "inference type things" to run. But really, we only have ONE inference type thing, or at least they will be provided via a singular plugin. 

Having a generalized "inference" thing in search/rank doesn't make sense. All the hooks for plugging in a nice rank feature action already exists. We should rely on that via the inference plugin.

Reply:
I see what you mean, that's fair - removed in ccf9c0f206cca6b20d3066967c0e6d5c2a6ca0b0